Figgle.Generator/RenderTextSourceGenerator.cs

```diff
@@ -97,244 +100,263 @@ public GenerateFiggleTextAttribute(string memberName, string fontName, string so
+                    if (!SyntaxFacts.IsValidIdentifier(generateFiggleInfo.MemberName))
+                    {
+                        context.ReportDiagnostic(Diagnostic.Create(
+                            InvalidMemberNameDiagnostic,
+                            generateFiggleInfo.AttributeLocation ?? targetType.Locations[0],
+                            generateFiggleInfo.MemberName ?? "unknown"));
+                        continue;
+                    }
 
-                    if (font is null)
+                    if (!memberNames.Add(generateFiggleInfo.MemberName!))
                     {
                         context.ReportDiagnostic(Diagnostic.Create(
-                            UnknownFontNameDiagnostic,
-                            item.FontNameLocation,
-                            item.FontName));
+                            DuplicateMemberNameDiagnostic,
+                            generateFiggleInfo.AttributeLocation ?? targetType.Locations[0],
+                            generateFiggleInfo.MemberName));
                         continue;
                     }
 
-                    if (!data.SeenMemberNames.Add(item.MemberName))
+                    if (targetType.GetMembers(generateFiggleInfo.MemberName!).Any())
                     {
                         context.ReportDiagnostic(Diagnostic.Create(
                             DuplicateMemberNameDiagnostic,
-                            item.MemberNameLocation,
-                            item.MemberName));
+                            targetType.Locations[0],
+                            generateFiggleInfo.MemberName));
                         continue;
                     }
 
-                    var text = font.Render(item.SourceText);
+                    if (string.IsNullOrWhiteSpace(generateFiggleInfo.FontName))
+                    {
+                        context.ReportDiagnostic(Diagnostic.Create(
+                            UnknownFontNameDiagnostic,
+                            targetType.Locations[0],
+                            generateFiggleInfo.MemberName ?? "unknown"));
+                        continue;
+                    }
 
-                    sb.AppendLine(
-                        $$"""
-                                public static string {{item.MemberName}} { get; } = @"{{text.Replace("\"", "\"\"")}}";
-                        """);
-                }
+                    var figgleFont = FiggleFonts.TryGetByName(generateFiggleInfo.FontName!);
+                    if (figgleFont is null)
+                    {
+                        // check if the requested font to use is available in the external fonts
+                        var matchingExternalFont = externalFonts.FirstOrDefault(
+                            externalFont => externalFont.FontName.Equals(
+                                generateFiggleInfo.FontName,
+                                StringComparison.OrdinalIgnoreCase));
```

Comment:
Nice! Should the README be updated to explain that external fonts work with both the generation and embedding source generators?